proto/catalog.proto

```diff
@@ -16,6 +16,13 @@ message ColumnIndex {
   uint64 index = 1;
 }
 
+message WatermarkDesc {
+  // The column idx the watermark is on
+  ColumnIndex watermark_idx = 1;
```

Comment:
I think if we do not need express the `Optional` field, we should not use `ColumnIndex` instead of uint or int.